.github/workflows/ci.yml

```diff
@@ -75,8 +75,8 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        python-version: ["3.7", "3.8", "3.9"]
-        nautobot-version: ["1.1.6", "stable"]
```

Comment:
I think you should always keep `stable` in the list even if it may be temporarily redundant so you can find bugs introduced by new versions of Nautobot.

Reply:
This gets in the way of the Dockerfile addition to allow local version override during the build. `stable` won't work with Poetry `add`.

Alternatively we could rename one of the vars but I wanted to keep it simple.

```
RUN poetry add nautobot=${NAUTOBOT_VER}
```

Reply:
Is that redundant with the invoke environment variables?
```
export INVOKE_NAUTOBOT_DEVICE_LIFECYCLE_MGMT_NAUTOBOT_VER=1.2.11
invoke build
```

Reply:
Problem is more with `NAUTOBOT_VER` being trumped by Poetry in the build process. This is a problem if you want to locally test against multiple versions.

When Docker builds image, after running `invoke build`, `pyproject.toml` will override Nautobot version in the base image.

So say `NAUTOBOT_VER` defines 1.2.10, and that's the image that's pulled, but then when package is installed Poetry might bump it to 1.3.10.

See similar pattern in GoldenConfig Dockerfile:

https://github.com/nautobot/nautobot-plugin-golden-config/blob/develop/development/Dockerfile#L32

Reply:
Ok, I understand the issue now - even in its current state it isn't working. I don't think the workaround it is very elegant, but as I don't currently have any suggestions for something better I will relent for now and resolve this conversation.

Reply:
In the SSOT plugin we recently added this pattern:

https://github.com/nautobot/nautobot-plugin-ssot/blob/develop/development/Dockerfile#L30

```docker
RUN if [ "$NAUTOBOT_VER" = "stable" ]; \
    then poetry add nautobot@latest; \
    else poetry add nautobot==$NAUTOBOT_VER; \
    fi
```

Reply:
That looks great @glennmatthews , I'm going to borrow this pattern!